src/AngryBytes/Hash/Hasher/Password.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Password.php
+ *
+ * @category        AngryBytes
+ * @package         Hash
+ * @subpackage      Hasher
+ * @copyright       Copyright (c) 2007-2016 Angry Bytes BV (http://www.angrybytes.com)
+ */
+
+namespace AngryBytes\Hash\Hasher;
+
+use AngryBytes\Hash\HasherInterface;
+
+use \InvalidArgumentException;
+use \RuntimeException;
+
+/**
+ * Password hasher
+ *
+ * Generate and verify hashes using the `password_*` functions.
+ * The hashing algorithm and salting is handled by these functions.
+ *
+ * @category        AngryBytes
+ * @package         Hash
+ * @subpackage      Hasher
+ */
+class Password implements HasherInterface
+{
+    /**
+     * The hashing algorithm
+     *
+     * @var int
+     */
+    const ALGORITHM = PASSWORD_DEFAULT;
+
+    /**
+     * Cost factor for the algorithm
+     *
+     * @var int
+     */
+    private $cost;
+
+    /**
+     * Get cost
+     *
+     * @return int
+     */
+    public function getCost()
+    {
+        return $this->cost;
+    }
+
+    /**
+     * Set cost
+     *
+     * @throws InvalidArgumentException if the cost is too high or low
+     * @param int|null $cost
+     * @return $this
+     */
+    public function setCost($cost)
+    {
+        if (is_null($cost)) {
```

Comment:
Waarom is dit toegestaan? Als het toegestaan is, dan moet je denk ik wel $this->cost overschrijven. De vraag is of we cost overschrijfbaar moeten maken nadat een hasher gemaakt is.